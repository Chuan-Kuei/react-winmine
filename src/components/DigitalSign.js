import React from "react";
import PropTypes from "prop-types";

import "./digitalSign.css";

const DigitalSign = ({ position, active }) => {
  const isActive = active ? "active" : "";
  return <div styleName={`container ${position} ${isActive}`} />;
};

DigitalSign.defaultProps = {
  active: false
};

DigitalSign.propTypes = {
  position: PropTypes.string.isRequired,
  active: PropTypes.bool
};

export default DigitalSign;
