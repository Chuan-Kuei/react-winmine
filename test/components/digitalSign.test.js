import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import DigitalSign from "../../src/components/DigitalSign";

describe.only("DigitalSign", () => {
  test("render correctly", () => {
    const wrapper = renderer
      .create(<DigitalSign position="left-top" active />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  test("props active is true and has class active", () => {
    const wrapper = shallow(<DigitalSign position="left-top" active />);
    expect(wrapper.hasClass("active")).toBe(true);
  });
});
