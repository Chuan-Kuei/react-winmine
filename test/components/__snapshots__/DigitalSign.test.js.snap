// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DigitalSign render correctly 1`] = `
<div
  className="container left-top vertical active"
/>
`;
